### PR TITLE
Cleanup install instructions

### LINKED ISSUE #198: Cannot use repo on MacOS / Apple / ARM / Silicon
I don't think this repository can be build on MacOs / Apple Sillicon

* NO GOAL: depends on Rosetta
* GOAL: Having the tooling "just works"
```
$ git clone https://github.com/dotnet/eShop

$ dotnet tool restore
# this repo is missing dotnet tool manifest for ef
$ dotnet new tool-manifest
$ dotnet tool install dotnet-ef
$ git add ./.config
$ git commit -m "Add required file for dotnet tool to avoid relying on a global computer state"

$ sudo dotnet workload restore
# dotnet workload is still bugged and require sudo on non-windows
# https://github.com/dotnet/sdk/issues/18585

$ dotnet restore eShop.Web.slnf
$ dotnet build eShop.Web.slnf
# should work
```

Current Result:
This repo depends on a package that is yet to be Apple ARM compliant:

```
dotnet build eShop.Web.slnf
MSBuild version 17.8.3+195e7f5a3 for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.

  Restore operation started

  4 libraries restored in 0,07 seconds

  EventBus -> ~/dev/github/eshop/src/EventBus/bin/Debug/net8.0/EventBus.dll
  Ordering.Domain -> ~/dev/github/eshop/src/Ordering.Domain/bin/Debug/net8.0/Ordering.Domain.dll
  eShop.ServiceDefaults -> ~/dev/github/eshop/src/eShop.ServiceDefaults/bin/Debug/net8.0/eShop.ServiceDefaults.dll
  WebAppComponents -> ~/dev/github/eshop/src/WebAppComponents/bin/Debug/net8.0/WebAppComponents.dll
  IntegrationEventLogEF -> ~/dev/github/eshop/src/IntegrationEventLogEF/bin/Debug/net8.0/IntegrationEventLogEF.dll
  Mobile.Bff.Shopping -> ~/dev/github/eshop/src/Mobile.Bff.Shopping/bin/Debug/net8.0/Mobile.Bff.Shopping.dll
  Ordering.Infrastructure -> ~/dev/github/eshop/src/Ordering.Infrastructure/bin/Debug/net8.0/Ordering.Infrastructure.dll
  EventBusRabbitMQ -> ~/dev/github/eshop/src/EventBusRabbitMQ/bin/Debug/net8.0/EventBusRabbitMQ.dll
  WebhookClient -> ~/dev/github/eshop/src/WebhookClient/bin/Debug/net8.0/WebhookClient.dll
~/.nuget/packages/grpc.tools/2.59.0/build/_protobuf/Google.Protobuf.Tools.targets(291,5): error MSB6003: The specified task executable "~/.nuget/packages/grpc.tools/2.59.0/tools/macosx_x64/protoc" could not be run. System.ComponentModel.Win32Exception (86): An error occurred trying to start process '~/.nuget/packages/grpc.tools/2.59.0/tools/macosx_x64/protoc' with working directory '~/dev/github/eshop/src/Basket.API'. Bad CPU type in executable [~/dev/github/eshop/src/Basket.API/Basket.API.csproj]
~/.nuget/packages/grpc.tools/2.59.0/build/_protobuf/Google.Protobuf.Tools.targets(291,5): error MSB6003:    at System.Diagnostics.Process.ForkAndExecProcess(ProcessStartInfo startInfo, String resolvedFilename, String[] argv, String[] envp, String cwd, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec) [~/dev/github/eshop/src/Basket.API/Basket.API.csproj]
~/.nuget/packages/grpc.tools/2.59.0/build/_protobuf/Google.Protobuf.Tools.targets(291,5): error MSB6003:    at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo) [~/dev/github/eshop/src/Basket.API/Basket.API.csproj]
~/.nuget/packages/grpc.tools/2.59.0/build/_protobuf/Google.Protobuf.Tools.targets(291,5): error MSB6003:    at Microsoft.Build.Utilities.ToolTask.ExecuteTool(String pathToTool, String responseFileCommands, String commandLineCommands) [~/dev/github/eshop/src/Basket.API/Basket.API.csproj]
~/.nuget/packages/grpc.tools/2.59.0/build/_protobuf/Google.Protobuf.Tools.targets(291,5): error MSB6003:    at Microsoft.Build.Utilities.ToolTask.Execute() [~/dev/github/eshop/src/Basket.API/Basket.API.csproj]
~/.nuget/packages/grpc.tools/2.59.0/build/_protobuf/Google.Protobuf.Tools.targets(291,5): error MSB6003: The specified task executable "~/.nuget/packages/grpc.tools/2.59.0/tools/macosx_x64/protoc" could not be run. System.ComponentModel.Win32Exception (86): An error occurred trying to start process '~/.nuget/packages/grpc.tools/2.59.0/tools/macosx_x64/protoc' with working directory '~/dev/github/eshop/src/WebApp'. Bad CPU type in executable [~/dev/github/eshop/src/WebApp/WebApp.csproj]
~/.nuget/packages/grpc.tools/2.59.0/build/_protobuf/Google.Protobuf.Tools.targets(291,5): error MSB6003:    at System.Diagnostics.Process.ForkAndExecProcess(ProcessStartInfo startInfo, String resolvedFilename, String[] argv, String[] envp, String cwd, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec) [~/dev/github/eshop/src/WebApp/WebApp.csproj]
~/.nuget/packages/grpc.tools/2.59.0/build/_protobuf/Google.Protobuf.Tools.targets(291,5): error MSB6003:    at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo) [~/dev/github/eshop/src/WebApp/WebApp.csproj]
~/.nuget/packages/grpc.tools/2.59.0/build/_protobuf/Google.Protobuf.Tools.targets(291,5): error MSB6003:    at Microsoft.Build.Utilities.ToolTask.ExecuteTool(String pathToTool, String responseFileCommands, String commandLineCommands) [~/dev/github/eshop/src/WebApp/WebApp.csproj]
~/.nuget/packages/grpc.tools/2.59.0/build/_protobuf/Google.Protobuf.Tools.targets(291,5): error MSB6003:    at Microsoft.Build.Utilities.ToolTask.Execute() [~/dev/github/eshop/src/WebApp/WebApp.csproj]
  Identity.API -> ~/dev/github/eshop/src/Identity.API/bin/Debug/net8.0/Identity.API.dll
  PaymentProcessor -> ~/dev/github/eshop/src/PaymentProcessor/bin/Debug/net8.0/PaymentProcessor.dll
  OrderProcessor -> ~/dev/github/eshop/src/OrderProcessor/bin/Debug/net8.0/OrderProcessor.dll
  Webhooks.API -> ~/dev/github/eshop/src/Webhooks.API/bin/Debug/net8.0/Webhooks.API.dll
  Ordering.API -> ~/dev/github/eshop/src/Ordering.API/bin/Debug/net8.0/Ordering.API.dll
  Catalog.API -> ~/dev/github/eshop/src/Catalog.API/bin/Debug/net8.0/Catalog.API.dll
  Ordering.UnitTests -> ~/dev/github/eshop/tests/Ordering.UnitTests/bin/Debug/net8.0/Ordering.UnitTests.dll
  Catalog.FunctionalTests -> ~/dev/github/eshop/tests/Catalog.FunctionalTests/bin/Debug/net8.0/Catalog.FunctionalTests.dll
  Ordering.FunctionalTests -> ~/dev/github/eshop/tests/Ordering.FunctionalTests/bin/Debug/net8.0/Ordering.FunctionalTests.dll

Build FAILED.

~/.nuget/packages/grpc.tools/2.59.0/build/_protobuf/Google.Protobuf.Tools.targets(291,5): error MSB6003: The specified task executable "~/.nuget/packages/grpc.tools/2.59.0/tools/macosx_x64/protoc" could not be run. System.ComponentModel.Win32Exception (86): An error occurred trying to start process '~/.nuget/packages/grpc.tools/2.59.0/tools/macosx_x64/protoc' with working directory '~/dev/github/eshop/src/Basket.API'. Bad CPU type in executable [~/dev/github/eshop/src/Basket.API/Basket.API.csproj]
~/.nuget/packages/grpc.tools/2.59.0/build/_protobuf/Google.Protobuf.Tools.targets(291,5): error MSB6003:    at System.Diagnostics.Process.ForkAndExecProcess(ProcessStartInfo startInfo, String resolvedFilename, String[] argv, String[] envp, String cwd, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec) [~/dev/github/eshop/src/Basket.API/Basket.API.csproj]
~/.nuget/packages/grpc.tools/2.59.0/build/_protobuf/Google.Protobuf.Tools.targets(291,5): error MSB6003:    at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo) [~/dev/github/eshop/src/Basket.API/Basket.API.csproj]
~/.nuget/packages/grpc.tools/2.59.0/build/_protobuf/Google.Protobuf.Tools.targets(291,5): error MSB6003:    at Microsoft.Build.Utilities.ToolTask.ExecuteTool(String pathToTool, String responseFileCommands, String commandLineCommands) [~/dev/github/eshop/src/Basket.API/Basket.API.csproj]
~/.nuget/packages/grpc.tools/2.59.0/build/_protobuf/Google.Protobuf.Tools.targets(291,5): error MSB6003:    at Microsoft.Build.Utilities.ToolTask.Execute() [~/dev/github/eshop/src/Basket.API/Basket.API.csproj]
~/.nuget/packages/grpc.tools/2.59.0/build/_protobuf/Google.Protobuf.Tools.targets(291,5): error MSB6003: The specified task executable "~/.nuget/packages/grpc.tools/2.59.0/tools/macosx_x64/protoc" could not be run. System.ComponentModel.Win32Exception (86): An error occurred trying to start process '~/.nuget/packages/grpc.tools/2.59.0/tools/macosx_x64/protoc' with working directory '~/dev/github/eshop/src/WebApp'. Bad CPU type in executable [~/dev/github/eshop/src/WebApp/WebApp.csproj]
~/.nuget/packages/grpc.tools/2.59.0/build/_protobuf/Google.Protobuf.Tools.targets(291,5): error MSB6003:    at System.Diagnostics.Process.ForkAndExecProcess(ProcessStartInfo startInfo, String resolvedFilename, String[] argv, String[] envp, String cwd, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec) [~/dev/github/eshop/src/WebApp/WebApp.csproj]
~/.nuget/packages/grpc.tools/2.59.0/build/_protobuf/Google.Protobuf.Tools.targets(291,5): error MSB6003:    at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo) [~/dev/github/eshop/src/WebApp/WebApp.csproj]
~/.nuget/packages/grpc.tools/2.59.0/build/_protobuf/Google.Protobuf.Tools.targets(291,5): error MSB6003:    at Microsoft.Build.Utilities.ToolTask.ExecuteTool(String pathToTool, String responseFileCommands, String commandLineCommands) [~/dev/github/eshop/src/WebApp/WebApp.csproj]
~/.nuget/packages/grpc.tools/2.59.0/build/_protobuf/Google.Protobuf.Tools.targets(291,5): error MSB6003:    at Microsoft.Build.Utilities.ToolTask.Execute() [~/dev/github/eshop/src/WebApp/WebApp.csproj]
    0 Warning(s)
    2 Error(s)

Time Elapsed 00:00:02.14
```

`Grpc.Tools` only ship for `macos x64` apparently, until Google actually update their package for Apple Silicon, should this repo offer an alternative way of using GRPC like for example `protobuff-net` ?

### LINKED ISSUE #299: Clarify prereqs / install steps in README.md
There are a few minor points we should clarify in the README:
- Users should install the latest _released_ version of the .NET 8 SDK. The link should point to install instructions for released SDKs, not nightlies
- Users should install the latest `azd`.
- Instructions for installing the aspire workload on Mac/Linux may require "sudo"

### LINKED ISSUE #355: Repo does not build under macOS via the command line
Tried to clone and build on the command line per instructions. Docker is running.
(macOS, Intel, .NET 8.0.204, Sonoma 14.4.1)

---
```
❯ git pull
Already up to date.
❯ dotnet restore eShop.Web.slnf
  Determining projects to restore...
  All projects are up-to-date for restore.
❯ dotnet run --project src/eShop.AppHost/eShop.AppHost.csproj
Building...
/Users/ahill/dev/dotnet/eShop/src/eShop.AppHost/Program.cs(22,47): error CS0234: The type or namespace name 'Identity_API' does not exist in the namespace 'Projects' (are you missing an assembly reference?) [/Users/ahill/dev/dotnet/eShop/src/eShop.AppHost/eShop.AppHost.csproj]
/Users/ahill/dev/dotnet/eShop/src/eShop.AppHost/Program.cs(28,45): error CS0234: The type or namespace name 'Basket_API' does not exist in the namespace 'Projects' (are you missing an assembly reference?) [/Users/ahill/dev/dotnet/eShop/src/eShop.AppHost/eShop.AppHost.csproj]
/Users/ahill/dev/dotnet/eShop/src/eShop.AppHost/Program.cs(33,46): error CS0234: The type or namespace name 'Catalog_API' does not exist in the namespace 'Projects' (are you missing an assembly reference?) [/Users/ahill/dev/dotnet/eShop/src/eShop.AppHost/eShop.AppHost.csproj]
/Users/ahill/dev/dotnet/eShop/src/eShop.AppHost/Program.cs(37,47): error CS0234: The type or namespace name 'Ordering_API' does not exist in the namespace 'Projects' (are you missing an assembly reference?) [/Users/ahill/dev/dotnet/eShop/src/eShop.AppHost/eShop.AppHost.csproj]
/Users/ahill/dev/dotnet/eShop/src/eShop.AppHost/Program.cs(42,29): error CS0234: The type or namespace name 'OrderProcessor' does not exist in the namespace 'Projects' (are you missing an assembly reference?) [/Users/ahill/dev/dotnet/eShop/src/eShop.AppHost/eShop.AppHost.csproj]
/Users/ahill/dev/dotnet/eShop/src/eShop.AppHost/Program.cs(46,29): error CS0234: The type or namespace name 'PaymentProcessor' does not exist in the namespace 'Projects' (are you missing an assembly reference?) [/Users/ahill/dev/dotnet/eShop/src/eShop.AppHost/eShop.AppHost.csproj]
/Users/ahill/dev/dotnet/eShop/src/eShop.AppHost/Program.cs(49,47): error CS0234: The type or namespace name 'Webhooks_API' does not exist in the namespace 'Projects' (are you missing an assembly reference?) [/Users/ahill/dev/dotnet/eShop/src/eShop.AppHost/eShop.AppHost.csproj]
/Users/ahill/dev/dotnet/eShop/src/eShop.AppHost/Program.cs(55,29): error CS0234: The type or namespace name 'Mobile_Bff_Shopping' does not exist in the namespace 'Projects' (are you missing an assembly reference?) [/Users/ahill/dev/dotnet/eShop/src/eShop.AppHost/eShop.AppHost.csproj]
/Users/ahill/dev/dotnet/eShop/src/eShop.AppHost/Program.cs(60,50): error CS0234: The type or namespace name 'WebhookClient' does not exist in the namespace 'Projects' (are you missing an assembly reference?) [/Users/ahill/dev/dotnet/eShop/src/eShop.AppHost/eShop.AppHost.csproj]
/Users/ahill/dev/dotnet/eShop/src/eShop.AppHost/Program.cs(64,42): error CS0234: The type or namespace name 'WebApp' does not exist in the namespace 'Projects' (are you missing an assembly reference?) [/Users/ahill/dev/dotnet/eShop/src/eShop.AppHost/eShop.AppHost.csproj]

The build failed. Fix the build errors and run again.
```

Can build and run an empty Aspire app. Default site opens in browser.

```❯ dotnet run apireTest.sln
Couldn't find a project to run. Ensure a project exists in /Users/ahill/dev/dotnet/apireTest, or pass the path to the project using --project.
❯ dotnet run --project apireTest.AppHost/apireTest.AppHost.csproj
Building...
info: Aspire.Dashboard.DashboardWebApplication[0]
      Now listening on: http://localhost:15054
info: Aspire.Dashboard.DashboardWebApplication[0]
      OTLP server running at: http://localhost:16171
warn: Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager[35]
      No XML encryptor configured. Key {ae4f76d1-029f-4584-bfc8-44fa812e1923} may be persisted to storage in unencrypted form.
info: Microsoft.Hosting.Lifetime[0]
      Application started. Press Ctrl+C to shut down.
info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Development
info: Microsoft.Hosting.Lifetime[0]
      Content root path: /Users/ahill/dev/dotnet/apireTest/apireTest.AppHost```
      
      
<img width="2328" alt="image" src="https://github.com/dotnet/eShop/assets/188638/c3049cc1-bbf9-4430-b49c-e72aa54388c6">

### PR DESCRIPTION
**Pull request by jamesmontemagno**
Fixes #355
Fixes #198
Fixes #299

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,18 +11,25 @@ A reference .NET application implementing an e-commerce website using a services
 ### Prerequisites
 
 - Clone the eShop repository: https://github.com/dotnet/eshop
-- (Windows only) Install Visual Studio. Visual Studio contains tooling support for .NET Aspire that you'll want to have. [Visual Studio 2022 version 17.10 Preview](https://visualstudio.microsoft.com/vs/preview/).
-  - During installation, ensure that the following are selected:
+- Install & start Docker Desktop: https://docs.docker.com/engine/install/
+
+#### Windows with Visutal Studio
+- Install [Visual Studio 2022 version 17.10 Preview](https://visualstudio.microsoft.com/vs/preview/).
```
Comment (danmoseley) [inline README.md:17]: we'll have to take off 'preview' in 2 weeks, you could drop it if you wanted I guess.

File: README.md [hunk 2]
```diff
@@ -11,18 +11,25 @@ A reference .NET application implementing an e-commerce website using a services
...
+  - Select the following workloads:
     - `ASP.NET and web development` workload.
     - `.NET Aspire SDK` component in `Individual components`.
+    - Optional: `.NET Muli-platform App UI development` to run client apps
```
Comment (danmoseley) [inline README.md:21]:
```suggestion
    - Optional: `.NET Multi-platform App UI development` to run client apps
```

File: README.md [hunk 3]
```diff
@@ -11,18 +11,25 @@ A reference .NET application implementing an e-commerce website using a services
...
+
+#### Mac, Linux, & Windows without Visual Studio
 - Install the latest [.NET 8 SDK](https://dot.net/download?cid=eshop)
-- On Mac/Linux (or if not using Visual Studio), install the Aspire workload with the following commands:
+- Install the .NET Aspire workload with the following commands:
```
Comment (danmoseley) [inline README.md:25]: do we have workload install troubleshooting somewhere to point to (@joperezr)?

File: README.md [hunk 4]
```diff
@@ -40,9 +47,11 @@ dotnet run --project src/eShop.AppHost/eShop.AppHost.csproj
 ```
 then look for lines like this in the console output in order to find the URL to open the Aspire dashboard:
 ```sh
-Now listening on: http://localhost:18848
+Login to the dashboard at: http://localhost:19888/login?t=uniquelogincodeforyou
 ```
 
+> You may need to install ASP.NET Core HTTPS development certificates first. Learn more at https://aka.ms/aspnet/https-trust-dev-cert
```
Comment (danmoseley) [inline README.md:53]:
```suggestion
> You may need to install ASP.NET Core HTTPS development certificates first, and then close all browser tabs. Learn more at https://aka.ms/aspnet/https-trust-dev-cert
```

**Review by danmoseley — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (jamesmontemagno) [reply README.md:17]: i'll wait ;)

File: README.md (hunk 3 above)
Comment (jamesmontemagno) [reply README.md:25]: I can point to: https://learn.microsoft.com/en-us/dotnet/aspire/fundamentals/setup-tooling?tabs=dotnet-cli%2Cunix#install-net-aspire

File: README.md [hunk 5]
```diff
@@ -11,18 +11,28 @@ A reference .NET application implementing an e-commerce website using a services
...
+- Install [Visual Studio 2022 version 17.10 Preview](https://visualstudio.microsoft.com/vs/preview/).
+  - Select the following workloads:
     - `ASP.NET and web development` workload.
     - `.NET Aspire SDK` component in `Individual components`.
+    - Optional: `.NET Multi-platform App UI development` to run client apps
+
+#### Mac, Linux, & Windows without Visual Studio
 - Install the latest [.NET 8 SDK](https://dot.net/download?cid=eshop)
-- On Mac/Linux (or if not using Visual Studio), install the Aspire workload with the following commands:
+- Install the .NET Aspire workload with the following commands:
```
Comment (jamesmontemagno) [inline README.md:25]:
```suggestion
- Install the [.NET Aspire workload](https://learn.microsoft.com/dotnet/aspire/fundamentals/setup-tooling?tabs=dotnet-cli%2Cunix#install-net-aspire) with the following commands:
```

File: README.md [hunk 6]
```diff
@@ -11,18 +11,28 @@ A reference .NET application implementing an e-commerce website using a services
...
+  - Select the following workloads:
     - `ASP.NET and web development` workload.
     - `.NET Aspire SDK` component in `Individual components`.
+    - Optional: `.NET Multi-platform App UI development` to run client apps
+
+#### Mac, Linux, & Windows without Visual Studio
 - Install the latest [.NET 8 SDK](https://dot.net/download?cid=eshop)
-- On Mac/Linux (or if not using Visual Studio), install the Aspire workload with the following commands:
+- Install the [.NET Aspire workload](https://learn.microsoft.com/dotnet/aspire/fundamentals/setup-tooling?tabs=dotnet-cli%2Cunix#install-net-aspire) with the following commands:
 ```powershell
```
Comment (tebeco) [inline README.md:26]: Won't work on `MacOs`
`dotnet workload` currently have an issue and require `sudo`

**Pull request merged**